### ISSUE TITLE
Zip for any quantity of sequences (In a lazy and static typed manner)

### ISSUE THREAD

**Issue by leandromoh** [morelinq/MoreLINQ#411]
LINQ only provide `Zip` for 2 sequences, if user need to zip any other quantity one can't do it so simply.

Based on Haskell's [ZipList](http://hackage.haskell.org/package/base-4.10.1.0/docs/Control-Applicative.html#v:ZipList) type, I created an operator, on my personal project, that can zip any quantity of sequences, In a completly lazy and static typed manner, as the following example shows.

```c#
Func<T1, T2, ..., TN, TResult> func = ....

IEnumerable<T1> seq1 = ...
IEnumerable<T2> seq2 = ...
.....
IEnumerable<TN> seqN = ...

IEnumerable<TResult> result = func.ZipMany(seq1)
                                  .ZipMany(seq2)
                                        ...
                                  .ZipMany(seqN);
```

Any interest on add this on MoreLinq?

**Comment by SuperJMN** [comment]
@atifaziz Even though the example that @leandromoh doesn't respond to the actual expected syntax, I have needed a Zip version that takes `IEnumerable<IEnumerable<T>>`

This this implementation by @ericlippert as a instance. 

https://stackoverflow.com/a/17977375/1025407

I really hope this gets implemented soon. `Zip`, `ZipLongest` and `ZipShortest` are needing this update a lot.

Sample scenario? This kata:
"Sums of Parts" https://www.codewars.com//kata/5ce399e0047a45001c853c2b/csharp

**Comment by SuperJMN** [comment]
@leandromoh Could you rephrase the scenario to reflect the issue better? You could take this post as an example: https://stackoverflow.com/questions/24181419/zip-multiple-abitrary-number-of-enumerables-in-c-sharp
